cohorts/functions.py

```diff
@@ -84,7 +84,8 @@ def count_filter_fn(filterable_variant, **kwargs):
             return ((filterable_variant_function(filterable_variant) if filterable_variant_function is not None else True) and
                     filter_fn(filterable_variant, **kwargs))
         patient_id = row["patient_id"]
-        return cohort.load_variants(
+        return cohort.load_effects(
```

Comment:
Yep, i did suspect that although it helps to have your confirmation :). Except I did run into some errors when using an `effect` filter on a filterable_variant. The first reason appeared to be that the parameter names are different (trivial to fix), whereas the second was an issue that the `filterable_variant.effect` object could not be found so a number of the later filters failed. 
At any rate I'm now:

1. validating that these are, indeed equivalent
2. testing that the filtered-cache-names are correct (ie equivalent when they should be, different for different filters).
3. Unifying the above (ie removing count_variants_function_builder, using count_effects_function_builder everywhere instead)

I'm not convinced that using an `only_snv` filter on the neoantigen_count will give us the response we want. But I need to do a little testing on this / review (maybe it's simpler than I think). `only_expressed` on that neoantigen_count will definitely need to be reworked, since the expressed_neoantigen_count is computed almost completely differently from the non-expressed count (the rationale for this is something I'd like to understand better before I mess with it).

One question/concern I am running into is the reason why "nonsynonymous" effects are handled differently from other effect-type filters.  it appears that, within varcode, we're filtering out `silent_and_noncoding` effects before picking the highest priority effect per variant. I would think the silent/noncoding effects would be the lowest-priority for a variant, of all possible effects so the order of this filtering wouldn't matter. In which case I'll make an `only_nonsynonymous` filter analogous to the rest. But curious to know your thoughts on this.

Secondly, and this is a related question that I've been thinking about now that I'm digging into this a bit more, any reason why we might want to filter effects prior to selecting the highest priority per variant? Ie is there a case where a variant could have one effect type that is not a subclass of its highest-priority effect? (ie where including all effects per variant & then filtering to include only X type would yield a different count from the current scenario where we keep only the highest priority & then filter to those where `effect.is_snv` is `True`?) [maybe this discussion should move into an issue, but it's relevant here .. so]